### PR TITLE
Update identity-readme.md

### PR DESCRIPTION
**Pull request by billwert**
Fixed various links to non-existent wiki pages.

There are still a bunch of dead links, specifically everything that is a "configuration" link in the tables under `Credential Classes`. It's not clear to me if it is better to keep the column there or rely on the other links in the document for that config info.

### CODE REVIEW

File: docs-ref-services/latest/identity-readme.md [hunk 1]
```diff
@@ -127,14 +127,14 @@ public void createDefaultAzureCredential() {
...
 
 ### Authenticating a user assigned managed identity with `DefaultAzureCredential`
 To Authenticate using User Assigned Managed Identity, please ensure that configuration instructions for your supported Azure Resource [here](#managed-identity-support) have been successfully completed.
 
 The below example demonstrates authenticating the `SecretClient` from the [azure-security-keyvault-secrets][secrets_client_library] client library using the `DefaultAzureCredential`, deployed to an Azure resource with a user assigned managed identity configured.
 
-See more about how to configure a user assigned managed identity for an Azure resource in [Enable managed identity for Azure resources](https://github.com/Azure/azure-sdk-for-java/wiki/Set-up-Your-Environment-for-Authentication#enable-managed-identity-for-azure-resources).
+See more about how to configure a user assigned managed identity for an Azure resource in [the managed identity documentation](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview).
```
Comment (joshfree) [inline docs-ref-services/latest/identity-readme.md:137]: all the en-us directories need to be dropped.  The site will auto-redirect to the correct locale

File: docs-ref-services/latest/identity-readme.md [hunk 2]
```diff
@@ -196,7 +194,7 @@ The [Managed identity authentication](https://docs.microsoft.com/azure/active-di
 ####  Authenticating in Azure with Managed Identity
 This examples demonstrates authenticating the `SecretClient` from the [azure-security-keyvault-secrets][secrets_client_library] client library using the `ManagedIdentityCredential` in a virtual machine, app service, function app, cloud shell, or AKS environment on Azure, with system assigned, or user assigned managed identity enabled.
 
-see more about how to configure your Azure resource for managed identity in [Enable managed identity for Azure resources](https://github.com/Azure/azure-sdk-for-java/wiki/Set-up-Your-Environment-for-Authentication#enable-managed-identity-for-azure-resources)
+see more about how to configure your Azure resource for managed identity in [the managed identity documentation](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview).
```
Comment (joshfree) [inline docs-ref-services/latest/identity-readme.md:197]: remove en-us/

**Review by joshfree — APPROVED** [review]

File: docs-ref-services/latest/identity-readme.md (hunk 2 above)
Comment (JasonWHowell) [reply docs-ref-services/latest/identity-readme.md]: Its better just to skip the domain name and start with slash (/azure/active-directory/managed-identities-azure-resources/overview)

File: docs-ref-services/latest/identity-readme.md (hunk 1 above)
Comment (JasonWHowell) [reply docs-ref-services/latest/identity-readme.md]: /azure/active-directory/managed-identities-azure-resources/overview

File: docs-ref-services/latest/identity-readme.md [hunk 3]
```diff
@@ -127,14 +127,14 @@ public void createDefaultAzureCredential() {
 }
 ```
 
-See more how to configure the `DefaultAzureCredential` on your workstation or Azure in [Configure DefaultAzureCredential](https://github.com/Azure/azure-sdk-for-java/wiki/Set-up-Your-Environment-for-Authentication#configure-defaultazurecredential).
+See more how to configure the `DefaultAzureCredential` on your workstation or Azure in [Authenticating with DefaultAzureCredential](https://github.com/Azure/azure-sdk-for-java/wiki/Azure-Identity-Examples#authenticating-with-defaultazurecredential).
```
Comment (Wlastiiq) [inline docs-ref-services/latest/identity-readme.md:130]: :) better

**Pull request closed without merging**